### PR TITLE
Community x

### LINKED ISSUE #111: Community Page - X Community
## Summary
Create a page that is routed to from the communities page for a specific community.
Show all the posts related to that community.

## Technical Details
Frontend
Use the post component for each post
Map works inside of rendering to map array elements to components
Remind me to create mock posts

## Acceptance Criteria
- Show all posts related to community
- Create a "Create Post" button
  - Does not need to be implemented here

## Figma
image

### PR DESCRIPTION
**Pull request by Aligary**
## Description
This PR creates the CommunityX page. CommunityX is the specific community that a user wants to interact with (Community 1 / Community 2) It displays the posts associated with that certain community, and allows users to create posts in this community. 

## Related Issue
Closes #111 
## Type of Changes
Please select the relevant option:
- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Documentation update
- [ ] Other (please specify)

## Things to Update
Please add any features, buttons, or pages that might need updating in the future:
- [ ] Button Functunality 
- [ ] Code Documentation 
- [ ] Mobile Responsiveness 
- [x] Code testing 
- [x] Other (please specify)

A lot of the data being displayed is dummy data. I'll make sure to check if anything is available to be requested in order to make this page more coherent. 
It is missing the following data: 
- Community Name
- Community Location
- Community Users
- Community Posts

## Screenshots 
Screenshot 2024-04-15 at 2 26 07 PM

## Additional Notes
If you look at main.tsx, we have this path. 
Screenshot 2024-04-15 at 2 41 59 PM
I know that the path to the community page should have its own specific route, but we do not have that set up yet. Once we are able to get this information, I will create a separate PR to make these changes. 

Also, should we include the navbar here?

### CODE REVIEW

**Review by dafneMena — APPROVED** [review]
Hi Aligary, great work. I believe Jordan's PR re the community get points has just been approved and can be merged. When he does so, you might be able to make some changes to the routing.

File: client/src/pages/CommunityX/CommunityX.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import { ArrowBackIos } from '@mui/icons-material';
+import { useNavigate } from "react-router-dom";
+
+/**
+ *  This function displays the CommunityX page. CommunityX is the specific community that a user selects to interact with.
+ */
+export default function CommunityX(): JSX.Element {
+
+    // replace this to get the posts corresponding to the community
+    // This is dummy data for now!!!
```
Comment (KingJordan152) [inline client/src/pages/CommunityX/CommunityX.tsx:13]: You can implement this now since #49 has been merged in

**Review by KingJordan152 — APPROVED** [review]
All of this looks really good. Great work, Ali!

Since #49 has been merged in, you can incorporate it in this PR, but since we're short on time you can omit this for now. In the meantime, approved!

**Review by Linkaloo — APPROVED** [review]

**Review by RedPenguin88 — APPROVED** [review]
Other than what Jordan noted, this looks good. Good use of comments should facilitate further development.

**Pull request merged**